File: 프로그래머스/0/181875. 배열에서 문자열 대소문자 변환하기/배열에서 문자열 대소문자 변환하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(vector<string> strArr) {
    vector<string> answer;
    //
    int idx = 0;
    for (auto& e : strArr)
    {
        for (auto& e2 : e)
        {
            if (idx % 2) e2 = toupper(e2);
            else e2 = tolower(e2);
        }
        answer.push_back(e);
        idx++;
    }
    //
    return answer;
}

File: 프로그래머스/0/181875. 배열에서 문자열 대소문자 변환하기/배열에서 문자열 대소문자 변환하기.java
import java.util.*;

class Solution {
    public String[] solution(String[] strArr) {
        ArrayList<String> list = new ArrayList<String>();
        for (int i = 0; i < strArr.length; i++) {
            String arr = "";
            if (i % 2 != 0) {
               
                for (int j = 0; j < strArr[i].length(); j++) {
                    arr += Character.toUpperCase(strArr[i].charAt(j));
                }
            } else {
                for (int j = 0; j < strArr[i].length(); j++) {
                    arr += Character.toLowerCase(strArr[i].charAt(j));
                }
            }
            list.add(arr);
        }
        String[] answer = new String[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}